File: sympy/axiom/discrete/sets/union/emptyset.py
from sympy.core.relational import Equality, LessThan, Unequality, \
    StrictGreaterThan, GreaterThan
from sympy.utility import plausible, Eq, Sum, Union, identity
from sympy.core.symbol import Symbol, dtype
from sympy.sets.sets import Interval
from sympy.axiom import discrete
from sympy import S
from sympy.concrete.expr_with_limits import Exists, Forall
from sympy.logic.boolalg import plausibles, And
from sympy.tensor.indexed import IndexedBase
from sympy.axiom.discrete.sets.union import inclusion_exclusion_principle
from sympy.axiom.discrete.sets.emptyset import strict_greater_than

# provided: A | B = {}
# A = {} and B = {}


def apply(provided):
    assert provided.is_Equality
    AB, emptyset = provided.args
    if emptyset != S.EmptySet:
        tmp = emptyset
        emptyset = AB
        AB = tmp
        assert emptyset == S.EmptySet

    assert AB.is_Union
    A, B = AB.args
    return And(Equality(A, S.EmptySet),
               Equality(B, S.EmptySet),
                  equivalent=provided,
                  plausible=plausible())


from sympy.utility import check


@check
def prove(Eq):
    A = Symbol('A', dtype=dtype.integer)
    B = Symbol('B', dtype=dtype.integer)

    equality = Equality(A | B, S.EmptySet)

    Eq << equality

    Eq << apply(equality)

    Eq << ~Eq[-1]

    Eq.A_nonempty, Eq.B_nonempty = Eq[-1].split()

    Eq.A_positive = Eq.A_nonempty.apply(discrete.sets.emptyset.strict_greater_than)

    Eq.B_positive = Eq.B_nonempty.apply(discrete.sets.emptyset.strict_greater_than)

    Eq.AB_union_empty = Eq[0].abs()

    Eq << Eq[0] - A

    Eq << Eq[-1].abs()

    Eq << Eq[-2].lhs.assertion()

    Eq << Eq[-1].subs(Eq[-2], Eq.AB_union_empty)

    Eq << Eq[-1].subs(Eq.A_positive)

    Eq << Eq[0] - B

    Eq << Eq[-1].abs()

    Eq << Eq[-2].lhs.assertion()

    Eq << Eq[-1].subs(Eq[-2], Eq.AB_union_empty)

    Eq << Eq[-1].subs(Eq.B_positive)

    Eq << ~Eq.A_nonempty
    Eq << ~Eq.B_nonempty

    Eq << (Eq[-1] & Eq[-2])


if __name__ == '__main__':
    prove(__file__)

File: axiom/discrete/sets/union/emptyset.php
<?php
require_once '..\..\..\..\utility.php';
$i = 0;
$txt[$i++] = "\\(A \\cup B = \\emptyset\\tag*{Eq[0]}\\)\\(A = \\emptyset\\wedge B = \\emptyset\\tag*{Eq[1]}\\)";
$txt[$i++] = "\\(\\exists_{A}{A \\neq \\emptyset} \\vee \\exists_{B}{B \\neq \\emptyset}\\tag*{~Eq[2]}\\)";
$txt[$i++] = "\\(\\exists_{A}{A \\neq \\emptyset}\\tag*{~Eq.A_nonempty}\\)";
$txt[$i++] = "\\(\\exists_{B}{B \\neq \\emptyset}\\tag*{~Eq.B_nonempty}\\)";
$txt[$i++] = "\\(\\exists_{A}{\\left|{A}\\right| > 0}\\tag*{~Eq.A_positive}\\)";
$txt[$i++] = "\\(\\exists_{B}{\\left|{B}\\right| > 0}\\tag*{~Eq.B_positive}\\)";
$txt[$i++] = "\\(\\left|{A \\cup B}\\right| = 0\\tag*{Eq.AB_union_empty}\\)";
$txt[$i++] = "\\(B \\setminus A = \\emptyset\\tag*{Eq[3]}\\)";
$txt[$i++] = "\\(\\left|{B \\setminus A}\\right| = 0\\tag*{Eq[4]}\\)";
$txt[$i++] = "\\(\\left|{A \\cup B}\\right| = \\left|{A}\\right| + \\left|{B \\setminus A}\\right|\\tag*{Eq[5]}\\)";
$txt[$i++] = "\\(0 = \\left|{A}\\right|\\tag*{Eq[6]}\\)";
$txt[$i++] = "\\(\\text{False}\\)";
$txt[$i++] = "\\(A \\setminus B = \\emptyset\\tag*{Eq[7]}\\)";
$txt[$i++] = "\\(\\left|{A \\setminus B}\\right| = 0\\tag*{Eq[8]}\\)";
$txt[$i++] = "\\(\\left|{A \\cup B}\\right| = \\left|{B}\\right| + \\left|{A \\setminus B}\\right|\\tag*{Eq[9]}\\)";
$txt[$i++] = "\\(0 = \\left|{B}\\right|\\tag*{Eq[10]}\\)";
$txt[$i++] = "\\(\\text{False}\\)";
$txt[$i++] = "\\(\\forall_{A}{A = \\emptyset}\\tag*{Eq[11]}\\)";
$txt[$i++] = "\\(\\forall_{B}{B = \\emptyset}\\tag*{Eq[12]}\\)";
$txt[$i++] = "\\(A = \\emptyset\\wedge B = \\emptyset\\tag*{Eq[1]}\\)";
render(__FILE__, $txt);
?>        

